packages/zoe/tools/types.js

```diff
@@ -0,0 +1,95 @@
+ * @property {(handler: TimerServiceHandler) => void} schedule Returns the time
+ * scheduled for the first call on handler.  The handler will continue to be
+ * scheduled for a `wake()` call every interval until the repeater is disabled.
+ * @property {() => void} disable Disable this repeater, so `schedule()` can't
+ * be called, and handlers already scheduled with this repeater won't be
+ * rescheduled again after `wake()` is next called on them.
+ */
+
+/**
+ * @typedef {Object} PriceQuote
+ * @property {Amount} quoteAmount Amount whose value is a PriceQuoteValue
+ * @property {Payment} quotePayment The `quoteAmount` wrapped as a payment
+ */
+
+/**
+ * @typedef {Array<PriceDescription>} PriceQuoteValue A set of PriceDescriptions
+ */
+
+/**
+ * @typedef {Object} PriceDescription A description of a single quote
+ * @property {Amount} amountIn The amount supplied to a trade
+ * @property {Amount} amountOut The quoted result of trading `amountIn`
+ * @property {TimerService} timer The service that gave the `timestamp`
+ * @property {Timestamp} timestamp A timestamp according to `timer` for the
+ * quote
+ * @property {any=} conditions Additional conditions for the quote
+ */
+
+/**
+ * @typedef {Object} PriceAuthority An object that mints PriceQuotes and handles
+ * triggers and notifiers for changes in the price
+ * @property {(brandIn: Brand, brandOut: Brand) => ERef<Issuer>} getQuoteIssuer
+ * Get the ERTP issuer of PriceQuotes for a given brandIn/brandOut pair
+ * @property {(brandIn: Brand, brandOut: Brand) => ERef<TimerService>}
+ * getTimerService get the timer used in PriceQuotes for a given
+ * brandIn/brandOut pair
+ * @property {(brandIn: Brand, brandOut: Brand) => ERef<Notifier<PriceQuote>>}
+ * getPriceNotifier
+ * @property {(deadline: Timestamp, amountIn: Amount, brandOut: Brand) =>
+ * Promise<PriceQuote>} quoteAtTime Resolves after `deadline` passes on the
+ * priceAuthority's timerService with the price quote of `amountIn` at that time
+ * @property {(amountIn: Amount, brandOut: Brand) => Promise<PriceQuote>}
+ * quoteGiven get a quote corresponding to the specified amountIn
+ * @property {(brandIn: Brand, amountOut: Amount) => Promise<PriceQuote>}
+ * quoteWanted get a quote corresponding to the specified amountOut
+ * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>}
+ * quoteWhenGT Resolve when a price quote of `amountIn` exceeds `amountOutLimit`
+ * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>}
+ * quoteWhenGTE Resolve when a price quote of `amountIn` reaches or exceeds
+ * `amountOutLimit`
+ * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>}
+ * quoteWhenLTE Resolve when a price quote of `amountIn` reaches or drops below
+ * `amountOutLimit`
+ * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>}
+ * quoteWhenLT Resolve when the price quote of `amountIn` drops below
+ * `amountOutLimit`
```

Comment:
I find this really hard to read. (I realize the primary consumer is the type system, but it's not the only one.) I think typescript allows whitespace. indenting successive lines of a single property would make this much easier for me to track. A lot of the difficulty is that the signatures are often broken across a newline. slightly longer lines would also help in a few cases, where only the method names go past 80 characters.

```suggestion
 * @typedef {Object} PriceAuthority An object that mints PriceQuotes and handles
 *   triggers and notifiers for changes in the price
 * @property {(brandIn: Brand, brandOut: Brand) => ERef<Issuer>} getQuoteIssuer
 *   Get the ERTP issuer of PriceQuotes for a given brandIn/brandOut pair
 * @property {(brandIn: Brand, brandOut: Brand) => ERef<TimerService>} getTimerService
 *   get the timer used in PriceQuotes for a given brandIn/brandOut pair
 * @property {(brandIn: Brand, brandOut: Brand) => ERef<Notifier<PriceQuote>>} getPriceNotifier
 * @property {(deadline: Timestamp, amountIn: Amount, brandOut: Brand) =>
 *   Promise<PriceQuote>} quoteAtTime Resolves after `deadline` passes on the
 *   priceAuthority's timerService with the price quote of `amountIn` at that time
 * @property {(amountIn: Amount, brandOut: Brand) => Promise<PriceQuote>} quoteGiven
 *   get a quote corresponding to the specified amountIn
 * @property {(brandIn: Brand, amountOut: Amount) => Promise<PriceQuote>} quoteWanted
 *   get a quote corresponding to the specified amountOut
 * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>} quoteWhenGT
 *   Resolve when a price quote of `amountIn` exceeds `amountOutLimit`
 * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>} quoteWhenGTE
 *   Resolve when a price quote of `amountIn` reaches or exceeds `amountOutLimit`
 * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>} quoteWhenLTE
 *   Resolve when a price quote of `amountIn` reaches or drops below `amountOutLimit`
 * @property {(amountIn: Amount, amountOutLimit: Amount) => Promise<PriceQuote>} quoteWhenLT
 *   Resolve when the price quote of `amountIn` drops below `amountOutLimit`
```

Reply:
I'm a victim of my 80-column-enforcing tooling.  I think it would be useful to put an extra blank line between the `@property` tags.  Hopefully that will be clear enough.

Reply:
Yes, that makes it very legible. Thanks.